src/workerd/api/tests/http-socket-test.js

```diff
@@ -0,0 +1,501 @@
+// Copyright (c) 2017-2024 Cloudflare, Inc.
+// Licensed under the Apache 2.0 license found in the LICENSE file or at:
+//     https://opensource.org/licenses/Apache-2.0
+
+import { connect, internalNewHttpClient } from 'cloudflare:sockets';
+import { strict as assert } from 'node:assert';
+import { setTimeout as sleep } from 'node:timers/promises';
+
+// Basic connectivity and GET test
+export const oneRequest = {
+  async test(ctrl, env, ctx) {
+    const socket = connect(`localhost:${env.HTTP_SOCKET_SERVER_PORT}`);
+    const httpClient = await internalNewHttpClient(socket);
```

Comment:
I may have missed it but there needs to be a test that checks if the socket cannot be established (e.g. connection refused) the `await internalNewHttpclient(...)` fails.

Reply:
Will addresss later.